crates/wasi-stub/src/lib.rs

```diff
@@ -137,24 +137,15 @@ pub fn stub_wasi_functions(
                                     ValType::Ref(r) => ValType::Ref(RefType {
                                         nullable: r.nullable,
                                         heap: match r.heap {
-                                            HeapType::Func => HeapType::Func,
-                                            HeapType::Extern => HeapType::Extern,
-                                            HeapType::Any => HeapType::Any,
-                                            HeapType::Eq => HeapType::Eq,
-                                            HeapType::Struct => HeapType::Struct,
-                                            HeapType::Array => HeapType::Array,
-                                            HeapType::I31 => HeapType::I31,
-                                            HeapType::NoFunc => HeapType::NoFunc,
-                                            HeapType::NoExtern => HeapType::NoExtern,
-                                            HeapType::None => HeapType::None,
-                                            HeapType::Index(index) => {
-                                                HeapType::Index(match index {
+                                            HeapType::Concrete(index) => {
+                                                HeapType::Concrete(match index {
                                                     Index::Num(n, s) => Index::Num(n, s),
                                                     Index::Id(id) => {
                                                         Index::Id(static_id(Some(id)).unwrap())
                                                     }
                                                 })
                                             }
+                                            HeapType::Abstract { shared, ty } => HeapType::Abstract { shared, ty },
```

Comment:
This is now simpler because most of the matches used to erase lifetime are now under `HeapType::Abstract` and are `Copy`.